### PR TITLE
Fixes #3643: Allow directory argument for Drupal phpunit tests.

### LINKED ISSUE #3643: Allow run-tests.sh script to be passed "directory" option.
**Is your feature request related to a problem? Please describe.**
The Drupal `run-tests.sh` script [allows](https://www.drupal.org/docs/8/phpunit/running-tests-through-command-line-with-run-testssh) for a `directory` option to be passed to limit where tests are searched for. I would like to get ALL `docroot/modules/custom` tests to run instead of passing in module names.

**Describe the solution you'd like**
Allow a `--directory` option to be passed to the script and specified in the `tests` BLT config key.

### PR DESCRIPTION
**Pull request by danepowell**
Fixes #3643
--------

Changes proposed
---------
- Support the directory option for run-tests.sh

Additional details
-----------
@dpagini and @greylabel , would you mind reviewing?

### CODE REVIEW

**Comment by greylabel** [comment]
I am trying to recall why I didn't add this in the first place...

File: src/Robo/Commands/Tests/RunTestsCommand.php [hunk 1]
```diff
@@ -115,6 +115,10 @@ public function executeTests() {
           $task->dieOnFail();
         }
 
+        if (isset($test['directory'])) {
+          $task->directory($test['directory']);
```
Comment (greylabel) [inline src/Robo/Commands/Tests/RunTestsCommand.php]: nit: the options in this file are arranged by `@param` type — maybe move this to where he other strings live.

File: src/Robo/Tasks/RunTestsTask.php [hunk 2]
```diff
@@ -134,6 +134,16 @@ public function dieOnFail() {
     return $this;
   }
 
+  /**
+   * @param string $directory
+   *
+   * @return $this
+   */
+  public function directory($directory) {
```
Comment (greylabel) [inline src/Robo/Tasks/RunTestsTask.php]: nit: the options in this file are arranged by `@param` type — maybe move this to where he other strings live.

**Comment by dpagini** [comment]
I pulled this patch down and it's working for me now.
Just as feedback I had to install java and the sqlite PHP drivers, and figure out a whole bunch of permissioning problems... but this overall is such a great step! Now that I have my `run-tests` working like I want, I'm going to attempt to switch over to using just PHPUnit.

**Comment by greylabel** [comment]
@dpagini Thanks for testing. I am always interested to hear more about any errors or issues you encountered along the way.

**Comment by danepowell** [comment]
> I am trying to recall why I didn't add this in the first place...

LOL I had the same thought... "if Grant didn't do this in the first place, there _must_ have been a good reason..."

**Pull request merged**